builtin/message_accumulator.go

```diff
@@ -0,0 +1,78 @@
+package builtin
```

Comment:
All of the files you are directly copying should include a comment near the top `// copied from <link to original file>` this makes review a lot easier as we can focus on the diffs that go beyond copying

Reply:
Yessss sorry! I'll rebase this tomorrow morning onto a commit that just copies the files, then you'll be able to see a commit with just the diff. Didn't have time today